packages/wallet/src/errors.ts

```diff
@@ -0,0 +1,55 @@
+import { inspect } from "node:util";
+
+export class InvalidPayloadError extends Error {
+    constructor(public payload: unknown) {
+        super(`Invalid payload: ${inspect(payload)}`);
+    }
+}
+
+export class RelayError extends Error {
+    constructor() {
+        super(
+            `You need to call the method relayDAppAddress from DAppAddressRelay__factory.`,
+        );
+    }
+}
+
+export class InsufficientBalanceError extends Error {
+    constructor(user: string, token: string, amount: bigint, tokenId?: bigint) {
+        super(
+            `insufficient balance of user ${user} of token ${token}: ${amount.toString()}${
+                tokenId ? ` of tokenId ${tokenId}` : ""
+            }`,
+        );
+    }
+}
+
+export class WalletUndefinedError extends Error {
+    constructor(address: string) {
+        super(`wallet of user ${address} is undefined`);
```

Comment:
are we raising this in what situation? Isn't a wallet always defined in the perspective of the user, even with a zero balance.

Reply:
In this situation, the user is asking to generate a voucher from a non-existent wallet. I can also change to generate an empty wallet. What do you think @fabiooshiro ?

Reply:
Ok removed.